File: btpy/src/btpy/btpy_loops/mod/has_some/has_some.py



def has_some(array:list|dict|str, function)->bool:
    """
    Función que itera sobre los 
    elementos del array enviado 
    buscando si alguno de los 
    elementos del array retornan 
    true a la función enviada.
    Se diferencia de has_all en que este
    se detiene al encontrar el valor
    indicado.
    """
    data = None
    if(type(array) == dict):
        data = array.values()
    else:
        data = array
    for e in data:
        if(function(e)):
            return True
    return False

File: btjs_page/btjs/mod/structures/has_some.js


import { Btjs } from "../../Btjs.js"

/*
Función que itera sobre los 
elementos del array enviado 
buscando si alguno de los 
elementos del array retornan 
true a la función enviada.
Se diferencia de has_all en que este
se detiene al encontrar el valor
indicado.
*/
export function has_some(ARRAY, 
        CHECK_FUNCTION) {
    let data = null
    if(Btjs.is_object_js(ARRAY)) {
        data = ARRAY.values()
    }
    else {
        data = ARRAY
    }
    for(let i in data) {
        if(CHECK_FUNCTION(data[i])) {
            return true
        }
    }
    return false
}
    